### PR TITLE
fix controller logic to prevent concurrent map reads and writes

### PR DESCRIPTION
**Pull request by wlan0**
There was a special case where under heavy load there was a concurrent map read and write. This fixes that

### CODE REVIEW

File: controller/controller.go [hunk 1]
```diff
@@ -387,21 +387,20 @@ func (c *ObjectStorageController) runController(ctx context.Context) {
...
 								}
 							}
-							c.opMap[uuid] = addOp{
+							c.opMap.Store(uuid, addOp{
 								Object:  d.Object,
 								AddFunc: &add,
 								Key:     key,
 								Indexer: indexer,
-							}
+							})
```
Comment (krishchow) [inline controller/controller.go:403]: Could we combine this by using `c.opMap.LoadOrStore` /nit

**Review by krishchow — COMMENTED** [review]
Minor question

File: controller/controller.go (hunk 1 above)
Comment (wlan0) [reply controller/controller.go]: Sure

**Pull request merged**